mage.go

```diff
@@ -0,0 +1,49 @@
+//go:build mage
+
+package main
+
+import (
+	"fmt"
+	"github.com/magefile/mage/sh"
+)
+
+const versionStringEnvVarName = "POCKET_VERSION_STRING"
+const pocketPackage = "github.com/pokt-network/pocket/cmd/pocket"
+const ldflag = "-X main.version=$" + versionStringEnvVarName
+
+var env = map[string]string{}
+
+func init() {
+	env["GO111MODULE"] = "on"
+
+	fmt.Println("Getting branch and commit information")
+	branch, _ := sh.Output("git", "branch", "--show-current")
+	hash, _ := sh.Output("git", "rev-parse", "--short", "HEAD")
+	_, dirty := sh.Output("git", "diff", "--quiet")
+
+	if branch != "" {
+		branch = "-" + branch
+		hash = "/" + hash
+	}
+	if dirty != nil && dirty.Error() == "running \"git diff --quiet\" failed with exit code 1" {
+		hash += "+dirty"
+	}
+	env[versionStringEnvVarName] = "0.0.0" + branch + hash
```

Comment:
NIT: I think the more idiomatic way to go about this is to use `fmt.Sprintf`. Example:

```
if branch != "" {
    fmt.Sprintf("0.0.0-%s/%s", branch, hash)
} else {
   ...
}
```

Reply:
I think this case might not fit the idiom since there are fragments of the version string that are optional. I will consider it, though, while I go through the code and contemplate the suggestions and restructure things.